plugin/go.vim

```diff
@@ -9,30 +9,22 @@ let s:cpo_save = &cpo
 set cpo&vim
 
 function! s:checkVersion() abort
-  " Not using the has('patch-7.4.2009') syntax because that wasn't added until
-  " 7.4.237, and we want to be sure this works for everyone (this is also why
-  " we're not using utils#EchoError()).
-  "
-  " Version 7.4.2009 was chosen because that's greater than what the most recent Ubuntu LTS
-  " release (16.04) uses and has a couple of features we need (e.g. execute()
-  " and :message clear).
-
   let l:unsupported = 0
   if go#config#VersionWarning() != 0
     if has('nvim')
       let l:unsupported = !has('nvim-0.3.2')
     else
-      let l:unsupported = (v:version < 704 || (v:version == 704 && !has('patch2009')))
+      let l:unsupported = v:version < 800
     endif
```

Comment:
Checking for just "Vim 8" isn't really enough, since it spans almost 2 years of development. You should pick a specific patch version like it did previously.

In the documentation above `8.0.1542` was chosen; why that version? The previous policy was always "what the latest Ubuntu LTS has", which is now Vim 8.0.1453 (18.04).

Reply:
Oh, I see now that you probably choose 8.0.1542 because that's in the Travis file? It has no special meaning, it just happened to be the version installed on my Arch laptop when I wrote that.

Reply:
We needed to pick something, and Vim8 is sufficient for now.

Reply:
So will Vim 8,0.0001 work? Probably not, since: https://github.com/fatih/vim-go/search?q=patch-8.0&unscoped_q=patch-8.0

These patch checks should probably also be removed, by the way. I don't know why it's such a fight to get all these details right every time... It's not even harder or anything :man_shrugging:

Reply:
My thinking here is that previously vim-go checked that it was at least Vim 7.4.2009. All we're doing is dropping support for Vim 7.4. Another patch to make sure it's at least 8.0.1453 makes sense. I'll add that later and probably drop the unneeded patch checks at the same time (if you're not motivated to submit it first), but didn't see any need to hold up the improvements that @rbisewski made.